File: examples/cesar.py
import re

def cipher(inpt, offset, charset):
    str = ""
    for c in inpt:
        if re.match(r".*\s.*", c):
            str += c
            continue
        j = charset.index(c)
        str += charset[(j + offset + len(charset)) % len(charset)]
    return str

File: examples/cesar.php
<?php

function cipher(string $input, int $offset, string $charset) {
    $str = "";
    for($i = 0; $i < strlen($input); $i++) {
        if(preg_match("/\s/", $input, $matches)) {
            $str .= $input[$i];
            continue;
        }
        $j = strpos($charset, $input[$i]);
        $str .= $charset[
            ($j + $offset + strlen($charset)) % strlen($charset)
        ];
    }
    return $str;
}